collectors/ciscoduo/ciscoduo_collector.js

```diff
@@ -0,0 +1,227 @@
+        const endTs = startTs.add(1, 'seconds');
+        const initialStates = [
+            {
+                duo_endpoint: DUO_ENDPOINT_AUTH,
+                mintime: startTs.valueOf(),
+                maxtime: endTs.valueOf(),
+                poll_interval_sec: 1
+            },
+            {
+                duo_endpoint: DUO_ENDPOINT_ADMIN,
+                mintime: startTs.valueOf(),
+                poll_interval_sec: 1
+            },
+            {
+                duo_endpoint: DUO_ENDPOINT_TELE,
+                mintime: startTs.valueOf(),
+                poll_interval_sec: 1
+            },
+            {
+                duo_endpoint: DUO_ENDPOINT_ENROLL,
+                mintime: startTs.valueOf(),
+                poll_interval_sec: 1
+            }
+        ];
+        return callback(null, initialState, 1);
+    }
+    
+    pawsGetLogs(state, callback) {
+        let collector = this;
+        console.info(`CDUO000001 Collecting data endpoint ${state.duo_endpoint} from ${state.mintime} till ${state.maxtime}`);
+        const client = new duo.Client(collector.clientId, collector.clientSecret, process.env.paws_endpoint);
+        let query = this._getDuoQuery(state);
+        
+        client.jsonApiCall('GET', state.duo_endpoint, query, function(res){
+            if (res.stat !== 'OK') {
+                console.error('CDUO000003 API call returned error: ' + res.message);
+                return callback(res.message);
+            } else {
+                return _handleDuoResults(res, callback);
+            }
+        });
+    }
+    
+    _getDuoQuery(state) {
+        let query;
+        
+        switch (state.duo_endpoint) {
+            case DUO_ENDPOINT_AUTH:
+                if (state.next_offset) {
+                    query = {
+                        next_offset: state.next_offset,
+                        limit: LIST_PAGE_SIZE
+                    };
+                } else {
+                    query = {
+                        mintime: state.mintime,
+                        maxtime: state.maxtime,
+                        limit: LIST_PAGE_SIZE
+                    };
+                }
+                break;
+            default:
+                query = {
+                    mintime: state.mintime
+                }
+                break;
+        }
+        return query;
+    }
+    
+    _handleDuoResults(res, callback) {
+        let logs;
+        let metadata;
+        let newState;
+        
+        switch (state.duo_endpoint) {
+            case DUO_ENDPOINT_AUTH:
+                logs = res.response.authlogs;
+                metadata = res.response.metadata;
+            case DUO_ENDPOINT_ADMIN:
+            case DUO_ENDPOINT_TELE:
+            case DUO_ENDPOINT_ENROLL:
+                logs = res.response;
+                // Assumption that the last message timestamp is a mintime for the next collection step
+                metadata = logs[logs.length-1];
```

Comment:
You need `break` statements after each case. otherwise it will keep moving down the stack even if it has already done a match.